### PR TITLE
[SCB-1139] update vertx version from 3.5.3 to 3.6.2

### PR DESCRIPTION
**Pull request by jeho0815**
1、update netty to 4.1.28 ncnative to 2.0.14.Final jackson 2.9.8
2、vertximpl and ConnectionBase(just for test) does not have an public construtor, so add the VertxImpl to resolve first, later will be removed
3、DefaultVertxMetrics does not have the vertx parameter, so add an setter method

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: transports/transport-rest/transport-rest-client/src/test/java/io/vertx/core/net/impl/ConnectionBase.java [hunk 1]
```diff
@@ -0,0 +1,398 @@
+/*
```
Comment (wujimin) [inline transports/transport-rest/transport-rest-client/src/test/java/io/vertx/core/net/impl/ConnectionBase.java]: must use vertx LICENSE
and why copy this file?

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterClient.java [hunk 2]
```diff
@@ -163,6 +163,7 @@ private void refreshMembers(MemberDiscovery memberDiscovery) {
       String configCenter = memberDiscovery.getConfigServer();
       IpPort ipPort = NetUtils.parseIpPortFromURI(configCenter);
       clientMgr.findThreadBindClientPool().runOnContext(client -> {
+        @SuppressWarnings("deprecation")
```
Comment (liubao68) [inline dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterClient.java]: Do not use deprecad api if necessary.

Comment (jeho0815) [reply dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterClient.java]: vertx 3.6.3 statement the interface is deprecated, but has not provider a new interface to replace, so we have to add the annotation to paas the CI buiding

File: transports/transport-rest/transport-rest-client/src/test/java/io/vertx/core/net/impl/ConnectionBase.java (hunk 1 above)
Comment (jeho0815) [reply transports/transport-rest/transport-rest-client/src/test/java/io/vertx/core/net/impl/ConnectionBase.java]: we mock some methods of ConnectionBase have changed to final, but mockito's the old version can't mock the final method, so copy this file.

File: LICENSE [hunk 3]
```diff
@@ -217,7 +217,10 @@ This product bundles files from vertx which is licensed under the Apache License
 For details, see https://github.com/vert-x3/vertx-web
 
 ================================================================
-For  foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/stream/PumpImplEx.java
+For transports/transport-rest/transport-rest-client/src/test/java/io/vertx/core/net/impl/ConnectionBase.java
```
Comment (wujimin) [inline LICENSE]: no need to declare ConnectionBase.java any more

File: pom.xml [hunk 4]
```diff
@@ -195,6 +195,8 @@
             <exclude>**/target/**</exclude>
             <!-- Skip the source files which are forked from vertx -->
             <exclude>**/io/vertx/ext/web/impl/MimeTypesUtils.java</exclude>
+            <exclude>**/io/vertx/core/net/impl/ConnectionBase.java</exclude>
```
Comment (wujimin) [inline pom.xml]: no need to declare ConnectionBase.java any more

**Pull request merged**